resource/graph_node.go

```diff
@@ -11,8 +11,30 @@ import (
 	"go.viam.com/rdk/logging"
 )
 
+//go:generate stringer -type NodeState -trimprefix NodeState
```

Comment:
Cool; I've never seen `go:generate` used before. Have you used it for enums before/why is it better than just writing a `String` method?

Reply:
Sorry missed this comment.

This stringer package basically writes the String method for you and ensures all enumerated cases are covered. It supports some basic transformations like trimming parts of the variable name or hard-coding a one-off string for a particular entry.

Reply:
Makes sense; I suppose valuable for enums if we add more variants in the future.